src/Features/Core/Portable/DocumentSpanExtensions.cs

```diff
@@ -28,9 +37,21 @@ public static bool TryNavigateTo(this DocumentSpan documentSpan, bool showInPrev
             var options = solution.Options.WithChangedOption(NavigationOptions.PreferProvisionalTab, showInPreviewTab);
             options = options.WithChangedOption(NavigationOptions.ActivateTab, activateTab);
 
+            return (workspace, service, options);
+        }
+
+        public static bool TryNavigateTo(this DocumentSpan documentSpan, bool showInPreviewTab, bool activateTab, CancellationToken cancellationToken)
```

Comment:
❔ Do we still need the synchronous forms here?

Reply:
would need to confirm with TS that they don't use it.

Reply:
although, i made these obsolete to help ensure they move off of them.